Don't pick up random source files (especially those under "PaxHeader" created
by pax-as-tar).

for ext in ['hpp', 'cpp']:
    for i in sorted(Path('.').glob(f'**/*.{ext}')): print(f'  {i}')
    print()

--- bindings/python/CMakeLists.txt.orig	2022-08-09 15:11:05.000000000 +0200
+++ bindings/python/CMakeLists.txt	2022-08-16 10:42:51.345631753 +0200
@@ -40,17 +40,111 @@
 SET(PYWRAP ${PROJECT_NAME}_pywrap)
 SET(PYWRAP ${PYWRAP} PARENT_SCOPE)
 
-FILE(GLOB_RECURSE ${PROJECT_NAME}_PYTHON_HEADERS
-     ${CMAKE_CURRENT_SOURCE_DIR}/*.hpp
+SET(${PROJECT_NAME}_PYTHON_HEADERS
+  algorithm/algorithms.hpp
+  fwd.hpp
+  multibody/data.hpp
+  multibody/fcl/transform.hpp
+  multibody/frame.hpp
+  multibody/geometry-data.hpp
+  multibody/geometry-model.hpp
+  multibody/geometry-object.hpp
+  multibody/joint/joint-derived.hpp
+  multibody/joint/joint.hpp
+  multibody/joint/joints-datas.hpp
+  multibody/joint/joints-models.hpp
+  multibody/joint/joints-variant.hpp
+  multibody/liegroups.hpp
+  multibody/model.hpp
+  multibody/pool/geometry.hpp
+  multibody/pool/model.hpp
+  parsers/python.hpp
+  parsers/sample-models.hpp
+  parsers/srdf.hpp
+  parsers/urdf.hpp
+  pybind11-all.hpp
+  pybind11.hpp
+  serialization/serializable.hpp
+  serialization/serialization.hpp
+  spatial/explog.hpp
+  spatial/force.hpp
+  spatial/inertia.hpp
+  spatial/motion.hpp
+  spatial/se3.hpp
+  utils/constant.hpp
+  utils/conversions.hpp
+  utils/copyable.hpp
+  utils/dependencies.hpp
+  utils/deprecation.hpp
+  utils/eigen.hpp
+  utils/list.hpp
+  utils/namespace.hpp
+  utils/pickle-map.hpp
+  utils/pickle-vector.hpp
+  utils/printable.hpp
+  utils/registration.hpp
+  utils/std-aligned-vector.hpp
+  utils/std-map.hpp
+  utils/std-vector.hpp
+  utils/version.hpp
      )
 
-FILE(GLOB_RECURSE ${PROJECT_NAME}_PYTHON_SOURCES
-     ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp
+SET(${PROJECT_NAME}_PYTHON_SOURCES
+  algorithm/expose-aba-derivatives.cpp
+  algorithm/expose-aba.cpp
+  algorithm/expose-algorithms.cpp
+  algorithm/expose-cat.cpp
+  algorithm/expose-centroidal-derivatives.cpp
+  algorithm/expose-centroidal.cpp
+  algorithm/expose-cholesky.cpp
+  algorithm/expose-com.cpp
+  algorithm/expose-contact-dynamics.cpp
+  algorithm/expose-crba.cpp
+  algorithm/expose-energy.cpp
+  algorithm/expose-frames-derivatives.cpp
+  algorithm/expose-frames.cpp
+  algorithm/expose-geometry.cpp
+  algorithm/expose-jacobian.cpp
+  algorithm/expose-joints.cpp
+  algorithm/expose-kinematic-regressor.cpp
+  algorithm/expose-kinematics-derivatives.cpp
+  algorithm/expose-kinematics.cpp
+  algorithm/expose-model.cpp
+  algorithm/expose-regressor.cpp
+  algorithm/expose-rnea-derivatives.cpp
+  algorithm/expose-rnea.cpp
+  algorithm/parallel/aba.cpp
+  algorithm/parallel/expose-parallel.cpp
+  algorithm/parallel/geometry.cpp
+  algorithm/parallel/rnea.cpp
+  math/expose-rpy.cpp
+  module.cpp
+  multibody/expose-data.cpp
+  multibody/expose-frame.cpp
+  multibody/expose-geometry.cpp
+  multibody/expose-liegroups.cpp
+  multibody/expose-model.cpp
+  multibody/fcl/expose-fcl.cpp
+  multibody/joint/expose-joints.cpp
+  multibody/pool/expose-pool.cpp
+  parsers/expose-parsers.cpp
+  parsers/python/model.cpp
+  parsers/sample-models.cpp
+  parsers/srdf.cpp
+  parsers/urdf/geometry.cpp
+  parsers/urdf/model.cpp
+  serialization/serialization.cpp
+  spatial/expose-SE3.cpp
+  spatial/expose-explog.cpp
+  spatial/expose-force.cpp
+  spatial/expose-inertia.cpp
+  spatial/expose-motion.cpp
+  spatial/expose-skew.cpp
+  utils/conversions.cpp
+  utils/dependencies.cpp
+  utils/version.cpp
      )
 
-REMOVE_PATH_FROM_LIST("${${PROJECT_NAME}_PYTHON_HEADERS}" "${CMAKE_CURRENT_SOURCE_DIR}/" ${PROJECT_NAME}_PYTHON_HEADERS)
-REMOVE_PATH_FROM_LIST("${${PROJECT_NAME}_PYTHON_SOURCES}" "${CMAKE_CURRENT_SOURCE_DIR}/" ${PROJECT_NAME}_PYTHON_SOURCES)
-
 # Parser for Python model
 IF(BUILD_WITH_LIBPYTHON)
   SET(${PROJECT_NAME}_PARSER_PYTHON_HEADERS
